### PR TITLE
CONSOLE-4075: Address tech debt in WebHookSecretForm component

### PR DESCRIPTION
**Pull request by nitin-dhevar**

### CODE REVIEW

**Comment by nitin-dhevar** [comment]
/hold

Depends on https://github.com/openshift/console/pull/14064

File: frontend/public/components/secrets/create-secret/WebHookSecretForm.tsx [hunk 1]
```diff
@@ -1,86 +1,48 @@
...
+          className="pf-v5-c-button pf-m-tertiary"
+          data-test="webhook-generate-button"
+        >
+          {t('public~Generate')}
+        </button>
+      </div>
+      <p className="help-block" id="webhook-secret-help">
+        {t('public~Value of the secret will be supplied when invoking the webhook.')}
+      </p>
+    </div>
```
Comment (TheRealJon) [inline frontend/public/components/secrets/create-secret/WebHookSecretForm.tsx:46]: Not new, but we are using 'webhook secret key' terminology here, when in reality this form controls the *value* of the webhook secret. The key is a static string: "WebhookSecretKey". We should update the wording.

File: frontend/public/components/secrets/create-secret/WebHookSecretForm.tsx [hunk 2]
```diff
@@ -1,86 +1,48 @@
...
-export const WebHookSecretForm = withTranslation()(WebHookSecretFormWithTranslation);
-
-type WebHookSecretFormState = {
-  stringData: {
-    [key: string]: string;
+export const WebHookSecretForm: React.FC<SecretSubFormProps> = ({ onChange, stringData }) => {
+  const { t } = useTranslation();
+  const changeWebHookSecretkey = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newSecret = event.target.value;
+    onChange({ stringData: { ...stringData, WebHookSecretKey: newSecret }, base64StringData: {} });
```
Comment (TheRealJon) [inline frontend/public/components/secrets/create-secret/WebHookSecretForm.tsx:9]: To keep this DRY, we could separate the onChange call into it's own handler, then the text input and generator button will just pass values into it.

**Review by TheRealJon — CHANGES_REQUESTED** [review]
Just a couple of nits.

**Comment by jhadvig** [comment]
Closing in favour of https://github.com/openshift/console/pull/14454

**Pull request closed without merging**